### PR TITLE
skipping tag notification if first time poster

### LINKED ISSUE #2408: Email recieved of an Unpublished node .
### Please describe the problem (or idea)

When a first time contributor on `publiclab.org` publishes a note , it has to be approved by the PublicLab moderators before it can appear to everyone . However an email is send to people who are subscribed to the TAGS that the note contain . 
On opening the link of note , we are redirected to the Dashboard .

### Solution

The email should be send to people after it gets approved by the maintainers .  

### What's your PublicLab.org username?

sagarpreet

### PR DESCRIPTION
**Pull request by Gauravano**
Fixes #2408 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in a uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] PR body includes `fixes #0000`-style reference to original issue #
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by Gauravano** [comment]
hey @jywarren, for now, I have disabled the notifications for tag addition on the unpublished note. I think we can have a follow-up on this figuring out how we can send emails to users after the first-timer-note is published. Thoughts?

**Comment by jywarren** [comment]
perhaps we should make an issue or plan for triggering tag notifications
upon approving, rather than upon publishing?

**Comment by Gauravano** [comment]
hey @jywarren, I have made some changes in `admin_controller.rb` so now tag notifications would be sent at time of post-approval for first-timer post. Please suggest if any changes are required. Thanks.

File: app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -168,6 +168,13 @@ def publish
           AdminMailer.notify_author_of_approval(@node, current_user).deliver_now
           AdminMailer.notify_moderators_of_approval(@node, current_user).deliver_now
           SubscriptionMailer.notify_node_creation(@node).deliver_now
+
+          if @node.tags.count > 0
+              @node.tags.each do |tag|
+                SubscriptionMailer.notify_tag_added(@node, tag, @node.author).deliver_now unless tag.subscriptions.empty?
```
Comment (jywarren) [inline app/controllers/admin_controller.rb]: This looks good, but the only issue is that people may receive multiple emails now. I want to suggest you look at how we collect emails in this PR (unfinished!) and send a single notification command to only unique emails (I had to make the email text a little more generic to accomplish this) -- https://github.com/publiclab/plots2/pull/2742/files

What do you think?

File: app/models/node.rb [hunk 2]
```diff
@@ -727,7 +727,9 @@ def add_tag(tagname, user)
                                  nid: id)
           if node_tag.save
             saved = true
-            SubscriptionMailer.notify_tag_added(self, tag, user).deliver_now unless tag.subscriptions.empty?
+            if status != 4
```
Comment (jywarren) [inline app/models/node.rb]: Perfect 👍 👍 👍

File: test/functional/tag_controller_test.rb [hunk 3]
```diff
@@ -562,4 +562,14 @@ def setup
     assert_equal true, assigns(:nodes).empty?
   end
 
+  test "do not notify if tag created on unpublished node" do
+    UserSession.create(users(:newcomer))
+
+    assert_difference 'ActionMailer::Base.deliveries.size', 0 do
```
Comment (jywarren) [inline test/functional/tag_controller_test.rb]: Love this!!!

File: app/controllers/admin_controller.rb (hunk 1 above)
Comment (Gauravano) [reply app/controllers/admin_controller.rb]: In our current system, we send a new mail for each tag addition like if a user is subscribed to three tags say `test1`,`test2`, `test3` , he will get 3 emails if a note is created with them. So, we need to develop a system where only  1 mail should be sent containing text such as `a note is generated with these 3 tags `. Am I getting it right?

**Comment by jywarren** [comment]
That's right -- and we should be able to just collect all recipients into
an array, then send one email to everyone once. Yep!!!

**Comment by Gauravano** [comment]
okay. In this PR, I will implement this for `first-timer note` but we need to open a separate issue for normal users as we are calling `add_tag()` for each tag and then from `add_tag` mail is send. I am opening a new issue about this and we can discuss for normal users there. What do you say?

**Comment by jywarren** [comment]
Ah, ok! Sounds good.

I think a slightly complicated but effective thing to do could be to look
over tags added in the past X period (say, 5 min or 10 min) and see who
would've been notified due to their current subscriptions, and NOT
re-notify if they should've received an email already due to a prior tag.

Can you copy in that idea? Thanks Gaurav!

**Comment by Gauravano** [comment]
yes @jywarren ! I will spend some time over this and let you know at which locations multiple emails are sent. Thanks.

**Comment by jywarren** [comment]
How's this going, @Gauravano ? I know it's a bit complex... Hope you're doing well!

**Comment by jywarren** [comment]
@Gauravano just checking in? Thanks! 👍

**Comment by Gauravano** [comment]
Sorry, I was a bit into Email Notification milestone. I will resolve it ASAP. Thanks!

**Comment by Gauravano** [comment]
Hi @jywarren please review this one! 
Previously, whenever someone creates a tag on a first-timer-note, a notification mail was sent but the user was unable to view the node as the node was in moderation queue. So, I have put a condition so mail wouldn't be set. And, once the node is approved then, mail is sent to all the tag subscribers too. What do you think? Thanks!

**Comment by Gauravano** [comment]
Yes, we can merge this. Thanks!

@jywarren is idea mentioned here https://github.com/publiclab/plots2/pull/2725#issuecomment-391882253 implemented as I have tried creating 3 different tags but received mail for one tag only ?

**Comment by jywarren** [comment]
I think it checks to see if a notification would have been sent due to an existing previous tag, and not send a new one if it's likely to have been sent.

**Pull request merged**